File: trees_and_graphs/print_all_nodes.java
public void printAllNodes(TreeNode root)
{
    Queue<TreeNode> queue = new LinkedList<>();
    queue.add(root);

    while (!queue.isEmpty())
    {
        int nodesInCurrentLevel = queue.size();
        
        // Do some logic here for the current level

        for (int i = 0; i < nodesInCurrentLevel; i++)
        {
            TreeNode node = queue.remove();

            // Do some logic here on the current node
            System.out.println(node.val);

            // Put the next level onto the queue
            if (node.left != null)
            {
                queue.add(node.left);
            }

            if (node.right != null)
            {
                queue.add(node.right);
            }
        }
    }
}

File: trees_and_graphs/print_all_nodes.py
from collections import deque

def print_all_nodes(root):
    queue = deque([root])
    while queue:
        nodes_in_current_level = len(queue)

        # Do some logic here for the current level

        for _ in range(nodes_in_current_level):
            node = queue.popleft()

            # Do some logic here on the current node
            print(node.val)

            # Put the next level onto the queue
            if node.left:
                queue.append(node.left)

            if node.right:
                queue.append(node.right)